### PR TITLE
LibJS: Convert "Operations on Objects" AOs to ThrowCompletionOr

### PR DESCRIPTION
**Pull request by linusg**
sorry Idan :sweat_smile: - revealed a handful of missing exception checks at least. Gonna do a test262 run before merging this, last time I would've regressed two things if I hadn't checked.

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Error.cpp [hunk 1]
```diff
@@ -36,12 +36,7 @@ ThrowCompletionOr<void> Error::install_error_cause(Value options)
     auto& vm = this->vm();
 
     // 1. If Type(options) is Object and ? HasProperty(options, "cause") is true, then
-    if (!options.is_object())
-        return {};
-    auto has_property = options.as_object().has_property(vm.names.cause);
-    if (auto* exception = vm.exception())
-        return throw_completion(exception->value());
-    if (has_property) {
+    if (options.is_object() && TRY(options.as_object().has_property(vm.names.cause))) {
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/Error.cpp:39]: :yakkiss:

File: Userland/Libraries/LibJS/Runtime/Object.cpp [hunk 2]
```diff
@@ -426,10 +420,7 @@ MarkedValueList Object::enumerable_own_property_names(PropertyKind kind) const
             // 2. Else,
 
             // a. Let value be ? Get(O, key).
-            auto value_or_error = get(property_name);
-            if (value_or_error.is_error())
-                return MarkedValueList { heap() };
-            auto value = value_or_error.release_value();
+            auto value = TRY(get(property_name));
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/Object.cpp:423]:
```c++
return { move(properties) };
```
Down below to please clang

**Review by IdanHo — APPROVED** [review]
Great work!

**Comment by linusg** [comment]
Accompanying test262 runner change (already merged): https://github.com/linusg/libjs-test262/commit/f0055c91c842227ed3ae357dc18509d6aac63d89

**Comment by linusg** [comment]
Should be good to go, but I'll add `MUST()` as discussed on Discord.

**Pull request merged**